src/Regression/Adapter/SugarCRMAware.php

```diff
@@ -409,4 +412,37 @@ public function shouldBeExecuted(): bool
         return (is_null($this->minVersion) || self::$sugarVersion['sugar_version'] >= $this->minVersion)
             && (is_null($this->maxVersion) || self::$sugarVersion['sugar_version'] <= $this->maxVersion);
     }
+
+    protected function detectSugarSubscription(): void
+    {
+        if ($this->config->getLicense() === Config::DEFAULT_LICENSE) {
+            $this
+                ->loginAs('admin')
+                ->bwcLogin()
+                ->send(
+                    new \GuzzleHttp\Psr7\Request(
+                        'GET',
+                        'index.php?module=Administration&action=LicenseSettings&bwcFrame=1'
+                    )
+                )
+                ->extractRegexp('licenseKey', '/name=\'license_key\' .* value="(.*?)"/')
+                ->logout();
+
+            $licenseKey = $this->getVar('licenseKey');
+        } else {
+            $licenseKey = $this->config->getLicense();
+        }
+
+        $subscriptionRequest = new Request('GET', "https://authenticate.sugarcrm.com/rest/subscription/$licenseKey");
+
+        $this
+            ->send($subscriptionRequest)
+            ->extract('subscriptionProduct', function (ResponseInterface $response) {
+                $data = json_decode((string)$response->getBody(), true);
+
+                return $data['subscription']['product'] ?? false;
+            });
+
+        static::$subscriptionProduct = $this->getVar('subscriptionProduct');
```

Comment:
It's not possible because in case if we return false in callback of method `extract` then we will receive an error.

`public function extract(
        string $variableName,
        callable $callback,
        ?string $errorMessage = null
    ): self {
        if (false === ($value = $callback($this->lastResponse))) {
            $this->throwException($errorMessage ?? 'Can not extract variable with the provided callback');
        }
        $this->status = Status::NO_ISSUE;
        $this->vars[$variableName] = $value;
        return $this;
    }`